lib/std/sha1.nim

```diff
@@ -213,7 +213,22 @@ proc secureHashFile*(filename: string): SecureHash =
   ## **See also:**
   ## * `secureHash proc <#secureHash,openArray[char]>`_ for generating a ``SecureHash`` from a string
   ## * `parseSecureHash proc <#parseSecureHash,string>`_ for converting a string ``hash`` to ``SecureHash``
-  secureHash(readFile(filename))
+  const BufferLength = 8192
+
+  let f = open(filename)
+  var state = newSha1State()
+  var buffer = newString(BufferLength)
+  while true:
+    let length = readChars(f, buffer, 0, BufferLength)
+    if length == 0:
+        break
+    buffer.setLen(length)
+    state.update(buffer)
+    if length != BufferLength:
+        buffer.setLen(BufferLength)
```

Comment:
```suggestion
      buffer.setLen(BufferLength)
```

Reply:
And also: Why? ``if length != BufferLength`` you're done, might as well break the loop.

Reply:
True, I will change.